pkgs/applications/science/logic/coq/default.nix

```diff
@@ -233,7 +234,8 @@ self = stdenv.mkDerivation {
 }; in
 if coqAtLeast "8.21" then self.overrideAttrs(o: {
   # coq-core is now a shim for rocq
-  propagatedBuildInputs = o.propagatedBuildInputs ++ [ rocq-core ];
+  propagatedBuildInputs = o.propagatedBuildInputs
+    ++ [ (rocq-core.override { version = o.exact-version; }) ];
```

Comment:
Why can't we just specify
```suggestion
    ++ [ (rocq-core.override { inherit version; }) ];
```
here?

Reply:
OK, so I expect this code would work?
```suggestion
    ++ [ (rocq-core.override { inherit (args) version; }) ];
```

But fine with your solution as well.

Reply:
Indeed, that would probably work too